server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -135,16 +135,15 @@ public long createSecret(String name, String encryptedSecret,
     SecretSeriesDAO secretSeriesDAO = secretSeriesDAOFactory.using(dslContext.configuration());
 
     Optional<SecretSeries> secretSeries = secretSeriesDAO.getSecretSeriesByName(name);
-    if (!secretSeries.isPresent()) {
-      return Optional.empty();
-    }
+    if (secretSeries.isPresent() && secretSeries.get().currentVersion().isPresent()) {
+      Optional<SecretContent> secretContent = secretContentDAO.getSecretContentBySecretIdOne(secretSeries.get().id());
+      if (!secretContent.isPresent()) {
+        return Optional.empty();
+      }
 
-    Optional<SecretContent> secretContent = secretContentDAO.getSecretContentBySecretIdOne(secretSeries.get().id());
-    if (!secretContent.isPresent()) {
-      return Optional.empty();
+      return Optional.of(SecretSeriesAndContent.of(secretSeries.get(), secretContent.get()));
     }
-
-    return Optional.of(SecretSeriesAndContent.of(secretSeries.get(), secretContent.get()));
+    return Optional.empty();
```

Comment:
Imho there are too many nested if's here so I'm not sure what the desired logic should be, but I think it looks like it does what I would expect.